src/vector/backends/awkward_constructors.py

```diff
@@ -256,11 +271,20 @@ def Array(*args: typing.Any, **kwargs: typing.Any) -> typing.Any:
     import vector
     import vector.backends.awkward  # noqa: 401
 
+    if isinstance(args[0], dict):
+        if all(isinstance(x, list) for x in args[0].values()):
+            raise TypeError(
+                "Use vector.zip to construct vectors using a list of coordinates"
+            )
+
     akarray = awkward.Array(*args, **kwargs)
     fields = awkward.fields(akarray)
 
     is_momentum, dimension, names, arrays = _check_names(akarray, fields)
 
+    if not _is_type_safe(akarray):
+        raise TypeError("A coordinate must be of the type int or float")
```

Comment:
Python style is for exception messages to be lower-cased (unlike this) and not end in punctuation (like this). I think there might be a PEP on that.